activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -86,29 +86,11 @@ def associated(*associations)
         @scope
       end
 
-      # Returns a new relation with left outer joins and where clause to identify
-      # missing relations.
-      #
-      # For example, posts that are missing a related author:
-      #
-      #    Post.where.missing(:author)
-      #    # SELECT "posts".* FROM "posts"
-      #    # LEFT OUTER JOIN "authors" ON "authors"."id" = "posts"."author_id"
-      #    # WHERE "authors"."id" IS NULL
-      #
-      # Additionally, multiple relations can be combined. This will return posts
-      # that are missing both an author and any comments:
-      #
-      #    Post.where.missing(:author, :comments)
-      #    # SELECT "posts".* FROM "posts"
-      #    # LEFT OUTER JOIN "authors" ON "authors"."id" = "posts"."author_id"
-      #    # LEFT OUTER JOIN "comments" ON "comments"."post_id" = "posts"."id"
-      #    # WHERE "authors"."id" IS NULL AND "comments"."id" IS NULL
       def missing(*associations)
         associations.each do |association|
           reflection = scope_association_reflection(association)
           @scope.left_outer_joins!(association)
-          if @scope.table_name == reflection.table_name
+          if @scope.values.size > 1
```

Comment:
This is definitely less complex and I think correct. I couldn't find an example where this has more than one scope value but wants `reflection.table_name`.

Does `associated` have the same bug? If so we should fix that in this PR too since it was changed in the other ones.

Reply:
Yes! It would have the same behavior introduced from the other bugs.  I will change that as well and add a test case for enum if it needs it.  Thanks for your time!

Reply:
Added this same logic to the `associated` method and added a test for it too.